base-ci-resource.yaml

```diff
@@ -104,6 +106,24 @@ testNodeApplicationUsingPostgres:
     POSTGRES_SERVER: postgres
     POSTGRES_PASSWORD: ''
 
+# If project is a node typescript app it will use this job configuration for build and testing
+testNodeTypescriptApplication:
+  extends: .node
+  image: briangweber/docker-node:carbon
+  stage: build
+  only:
+    variables:
+      - $NODEPOSTGRESAPP == null
```

Comment:
These compound variables are what is getting us into  the not-sharable blocks of build steps. Technically, you now want NODE_Typescript_postgres for the permutations there.

Would just a `TYPESCRIPT && npm run build` in the regular node build work?

Reply:
that would be nice, i'm not a fan of all the variables either

Reply:
oh awesome it worked: https://cimpress.githost.io/commerce/network-controller/transaction-coordinator/-/jobs/970837 I can see the tsc command

Reply:
have to backpedal, https://cimpress.githost.io/commerce/notifications-hub/subscriptions/-/jobs/970848
non typescript projects won't pass with this